src/main/java/dev/lambdaurora/spruceui/widget/AbstractSpruceButtonWidget.java

```diff
@@ -137,60 +138,60 @@ protected int getVOffset() {
 	}
 
 	@Override
-	protected void renderWidget(MatrixStack matrices, int mouseX, int mouseY, float delta) {
-		this.renderButton(matrices, mouseX, mouseY, delta);
+	protected void renderWidget(GuiGraphics guiGraphics, int mouseX, int mouseY, float delta) {
+		this.renderButton(guiGraphics, mouseX, mouseY, delta);
 		if (!this.dragging)
 			Tooltip.queueFor(this, mouseX, mouseY, this.tooltipTicks,
 					i -> this.tooltipTicks = i, this.lastTick, i -> this.lastTick = i);
 	}
 
-	protected void renderButton(MatrixStack matrices, int mouseX, int mouseY, float delta) {
+	protected void renderButton(GuiGraphics guiGraphics, int mouseX, int mouseY, float delta) {
 		int color = this.active ? 16777215 : 10526880;
-		drawCenteredText(matrices, this.client.textRenderer, this.getMessage(),
+		guiGraphics.drawCenteredShadowedText(this.client.textRenderer, this.getMessage(),
 				this.getX() + this.getWidth() / 2, this.getY() + (this.getHeight() - 8) / 2,
 				color | MathHelper.ceil(this.alpha * 255.0F) << 24);
 	}
 
 	@Override
-	protected void renderBackground(MatrixStack matrices, int mouseX, int mouseY, float delta) {
+	protected void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float delta) {
 		RenderSystem.setShaderColor(1.f, 1.f, 1.f, this.getAlpha());
 		RenderSystem.setShaderTexture(0, ClickableWidget.WIDGETS_TEXTURE);
```

Comment:
Is this even still necessary if the drawTexture method now takes a texture identifier?

Reply:
Technically, it is not, but it still ensures it is changed.